fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/aggregation/FilteredAggregation.java

```diff
@@ -59,6 +59,7 @@ public FilteredAggregation(@NotNull String name, String fieldName, Aggregation a
     }
 
     /**
+     * TODO maybe rewrite to reflect logic change
      * Splits an Aggregation for 2-pass aggregation into an inner filtered aggregation &amp; outer aggregation. The
      * outer aggregation is obtained by unwrapping the inner filtered aggregation and getting just the aggregation.
      * The outer aggregation fieldName will reference the inner aggregation name. The inner aggregation is unmodified.
```

Comment:
Yeah, rewrite:

Filtered aggregation delegates nesting rules to the aggregation being filtered.  Then the filter is applied to the inner aggregation which is both probably efficient on the size of the aggregation performed, and also means that the filtering dimension doesn't need to be added to the grouping expression of the nested query.